### PR TITLE
Support for flows, subflows and executions

### PR DESCRIPTION
**Pull request by tomrutsaert**
support for flows, subflows and executions
(+ added missing fields in data_source_keycloak_realm)

This are the changes form PR [#138](https://github.com/mrparkers/terraform-provider-keycloak/pull/138) + adaptations needed for keycloak 8.0.x 

See comments in #138 for more info

### CODE REVIEW

**Comment by mrparkers** [comment]
Thanks for re-opening this, I'll have some time later this week to review this again :smile:

File: provider/data_source_keycloak_realm.go [hunk 1]
```diff
@@ -60,6 +60,10 @@ func dataSourceKeycloakRealm() *schema.Resource {
 				Type:     schema.TypeBool,
 				Computed: true,
 			},
+			"ssl_required": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
```
Comment (tomrutsaert) [inline provider/data_source_keycloak_realm.go]: Changes in this file are not really related to the PR, but I saw they were missing, so I added them. see also #218

File: keycloak/authentication_subflow.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+func (keycloakClient *KeycloakClient) getExecutionId(authenticationSubFlow *AuthenticationSubFlow) (string, error) {
+	list, err := keycloakClient.ListAuthenticationExecutions(authenticationSubFlow.RealmId, authenticationSubFlow.ParentFlowAlias)
+	if err != nil {
+		return "", err
+	}
+
+	var executionId = ""
+	for _, ex := range list {
+		if ex.FlowId == authenticationSubFlow.Id {
+			executionId = ex.Id
```
Comment (mrparkers) [inline keycloak/authentication_subflow.go]: you could return here instead of breaking

File: keycloak/authentication_subflow.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+			break
+		}
+	}
+	if executionId == "" {
+		return "", errors.New("no execution id found for subflow")
+	}
+	return executionId, nil
```
Comment (mrparkers) [inline keycloak/authentication_subflow.go]: if you return in the for loop, then you can assume this fails if the code reaches this point

File: keycloak/authentication_execution.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+func (keycloakClient *KeycloakClient) RaiseAuthenticationExecutionPriority(realmId, id string) error {
+	_, _, err := keycloakClient.post(fmt.Sprintf("/realms/%s/authentication/executions/%s/raise-priority", realmId, id), nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (keycloakClient *KeycloakClient) LowerAuthenticationExecutionPriority(realmId, id string) error {
```
Comment (mrparkers) [inline keycloak/authentication_execution.go]: are these two functions used?  if not, how should users raise and lower the priority for executions?

Comment (tomrutsaert) [reply keycloak/authentication_execution.go]: There are not used, but they do work.
I left in here in the case some one worked out a better solution, and could use these methods....

Should I remove them?

File: keycloak/authentication_subflow.go (hunk 2 above)
Comment (tomrutsaert) [reply keycloak/authentication_subflow.go]: I will change this.

File: keycloak/authentication_execution.go (hunk 4 above)
Comment (mrparkers) [reply keycloak/authentication_execution.go]: No, I was just curious how the priority was determined, but then I realized in your example that you used `depends_on` in order to ensure subflows with higher priorities were created before subflows with lower priorities.

This approach makes sense, although it would be impossible to update this aftwards.  But if that's good enough for you then I'm okay with this.  I'm hoping that this is good enough for most users until Keycloak X is released, in which case I'll probably have to rewrite a lot of this provider anyways.
